c/json.c

```diff
@@ -1231,6 +1287,59 @@ JsonToken *getNumberToken(JsonTokenizer *tokenizer) {
   return makeJsonToken(tokenizer, badNumber ? JSON_TOKEN_BAD_NUMBER : JSON_TOKEN_NUMBER, text);
 }
 
+static int addTokenChar(JsonTokenizer *tokenizer, int pos, int c){
+  if (pos < tokenizer->bufferSize){
+    tokenizer->buffer[pos++] = c;
+  }
+  return pos;
+}
+
+static 
+JsonToken *getNumberTokenV2(JsonTokenizer *tokenizer) {
+  char *buffer = tokenizer->buffer;
+  int pos = 0;
+  bool isInteger = true;
+  bool badNumber = false;
+  
+  if (jsonTokenizerLookahead(tokenizer) == '-') {
+    buffer[pos++] = jsonTokenizerRead(tokenizer);
+  }
+  if (isdigit(jsonTokenizerLookahead(tokenizer))) {
+    buffer[pos++] = jsonTokenizerRead(tokenizer);
+    while (isdigit(jsonTokenizerLookahead(tokenizer))) {
+      int c = jsonTokenizerRead(tokenizer);
+      pos = addTokenChar(tokenizer,pos,c);
+    }
+    if (jsonTokenizerLookahead(tokenizer) == '.'){
+      isInteger = false;
+      pos = addTokenChar(tokenizer,pos,jsonTokenizerRead(tokenizer));
+      if (isdigit(jsonTokenizerLookahead(tokenizer))){
+        pos = addTokenChar(tokenizer,pos,jsonTokenizerRead(tokenizer));
+        while (isdigit(jsonTokenizerLookahead(tokenizer))) {
+          int c = jsonTokenizerRead(tokenizer);
+          pos = addTokenChar(tokenizer,pos,c);
+        }
+      } else {
+        badNumber = true;
+      }
+    }
+    /* must have at least one number, followed by a lookahead to a non-number here */
+  } else {
+    badNumber = true;
+    int c = jsonTokenizerRead(tokenizer);
+    pos = addTokenChar(tokenizer,pos,c);
+  }
+  char *text = jsonTokenizerAlloc(tokenizer, pos + 1);
+  memcpy(text, buffer, pos);
+  if (badNumber){
+    return makeJsonToken(tokenizer,JSON_TOKEN_BAD_NUMBER,text);
```

Comment:
I know it's probably not that important, but we should probably look at ways to automate style/coding guideline checks (e.g. this code looks different from the existing code here). It can be done via clang-format, and it'd be easy to tell committers how things should be formatted by just asking to use clang-format with our config file.

Reply:
I agree not in the scope of this PR